src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Data.SqlClient.SNI
+{
+    /// <summary>
+    /// SNI Asynchronous callback
+    /// </summary>
+    /// <param name="packet">SNI packet</param>
+    /// <param name="sniErrorCode">SNI error code</param>
+    internal delegate void SNIAsyncCallback(SNIPacket packet, uint sniErrorCode);
```

Comment:
We use this form of delegate in a lot places, so using a more descriptive type helps with readability. Instead of having to parse out what the Action does, you know it's supposed to be a callback.